# Analysis of Tags for Table Tennis Leipzig


In [11]:
import os
import json

In [12]:
current_dir = os.getcwd()
data_dir = os.path.abspath(os.path.join(current_dir, '..', 'Leipzig_table_tennis_locations.json'))
data_dir

'/home/claraleidhold/spatialhumanities/ball-bounce-buddies/data/Leipzig_table_tennis_locations.json'

In [13]:
with open(data_dir,'r') as json_file:
    data = json.load(json_file)

In [14]:
location_id_count = 0
for _, _ in data.items():
    location_id_count += 1

print('There are %d different Table Tennis Locations in Leipzig' % location_id_count)


There are 143 different Table Tennis Locations in Leipzig


## Tag Overview: Location Description

For every Location ID there is a location description, location latitude and longitude existing. 
Let's discover the location description by analysing its tags:

In [15]:
tag_stats = {}
for location_info in data.values():

    tags = location_info.get('tags', {})
    
    for tag, value in tags.items():
        if tag not in tag_stats:
            tag_stats[tag] = {'count': 1, 'values': {value:1}}
        else:
            tag_stats[tag]['count'] += 1
            if value not in tag_stats[tag]['values']:
                tag_stats[tag]['values'][value] = 1
            else:
                tag_stats[tag]['values'][value] += 1


In [16]:
unique_tags = 0
for tag, _ in tag_stats.items():
    unique_tags += 1
print('There are %d unique tags for these locations.' % unique_tags)

There are 24 unique tags for these locations.


This shows the tag appearance out of 143 times.

In [17]:
for key, _ in tag_stats.items():
    count = tag_stats.get(key,{}).get('count')
    print('%s: %d ' % (key,count))

check_date: 21 
leisure: 137 
note: 1 
sport: 143 
wheelchair: 5 
access: 29 
lit: 21 
surface: 1 
name: 2 
opening_hours: 1 
shop: 1 
capacity: 6 
covered: 6 
fee: 3 
net: 3 
net:material: 3 
net:overhang: 3 
playability: 3 
material: 3 
start_date: 1 
outdoor: 1 
addr:city: 1 
addr:postcode: 1 
addr:street: 1 


In [51]:
print('Tag statistics:')
for tag, info in tag_stats.items():
    count = info['count']
    unique_values = info['values']
    print(f"Tag: {tag}, Count: {count}, Values: {unique_values}")


Tag statistics:
Tag: check_date, Count: 21, Values: {'2022-12-21': 1, '2021-01-11': 1, '2021-09-14': 1, '2022-06-21': 2, '2022-02-03': 1, '2021-03-07': 1, '2021-04-02': 1, '2021-01-19': 1, '2020-12-30': 1, '2021-02-10': 1, '2022-08-08': 1, '2021-02-27': 1, '2022-03-31': 1, '2022-01-17': 1, '2023-05-17': 2, '2022-07-08': 1, '2023-03-16': 1, '2023-02-17': 1, '2023-05-20': 1}
Tag: leisure, Count: 137, Values: {'pitch': 136, 'playground': 1}
Tag: note, Count: 1, Values: {'1 table': 1}
Tag: sport, Count: 143, Values: {'table_tennis': 143}
Tag: wheelchair, Count: 5, Values: {'yes': 4, 'limited': 1}
Tag: access, Count: 29, Values: {'yes': 20, 'private': 6, 'customers': 2, 'permissive': 1}
Tag: lit, Count: 21, Values: {'no': 21}
Tag: surface, Count: 1, Values: {'paved': 1}
Tag: name, Count: 2, Values: {'TT Shop Leipzig': 1, 'Tischtennisplatz': 1}
Tag: opening_hours, Count: 1, Values: {'Tu-Fr 09:00-12:30,14:30-18:00; Sa 09:00-12:30': 1}
Tag: shop, Count: 1, Values: {'sports': 1}
Tag: capacity, 